app/api/notify-callback/route.ts

```diff
@@ -136,9 +136,9 @@ export const POST = middleware([validAuthorizationHeader()], async (req, props)
     });
   }
 
-  if (deliveryStatus === "permanent-failure") {
+  if (deliveryStatus === "technical-failure" || deliveryStatus === "permanent-failure") {
```

Comment:
Just a comment but another pattern that allows for more statuses to be added in the future could be done like this:
```js
const terminalFailures = ["permanent-failure","technical-failure"];
if (terminalFailures.includes(deliveryStatus)){
// do not retry
}
```